homeassistant/setup.py

```diff
@@ -240,12 +258,26 @@ def log_error(msg: str) -> None:
     elif name in processed:
         return
 
-    if hasattr(module, 'DEPENDENCIES'):
-        dep_success = await _async_process_dependencies(
-            hass, config, name, module.DEPENDENCIES)  # type: ignore
+    manifest = await hass.async_add_executor_job(
+        json_util.load_json,
+        os.path.join(os.path.dirname(module.__file__), 'manifest.json')
+    )
 
-        if not dep_success:
-            raise HomeAssistantError("Could not set up all dependencies.")
+    if manifest and manifest['dependencies']:
```

Comment:
Please remove `_async_process_deps` and instead do something like this:

```python
deps = manifest['dependencies'] if manifest else getattr(module, 'DEPENDENCIES', [])

if deps and not await _async_process_dependencies(…):
    raise HomeAssistantError("Could not set up all dependencies.")
```

Reply:
Sorry, I misunderstood your original comment, should be fixed now.